async.go

```diff
@@ -90,3 +105,52 @@ func panicHandler(ctx context.Context, errCh chan error) {
 func getContextWithTaskName(ctx context.Context, taskName string) context.Context {
 	return loggers.AddToLogContext(ctx, "taskname", taskName)
 }
+
+func GoWithWait(ctx context.Context,
+	taskNames []string,
+	tasks []func(context.Context) error,
+	timeout time.Duration,
+	logger log.Logger) error {
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer func() {
+		if ctx.Err() == nil {
+			cancel()
+		}
+	}()
+
+	errs := make(chan error, len(tasks))
```

Comment:
good to add one more size, to be on safer side in case we get multiple error from the single routine, Should not happen, +1 to the len wouldnt be bad either.